### ISSUE TITLE
Failure of wf 20434.1 after Phase2 relval updates

### ISSUE THREAD

**Issue by fabiocos** [cms-sw/cmssw#27507]
After #27449 that has update the Phase2 relval sets in the matrix, test wf 20434.1 fails in a reproducible way. This is a D41 scenario workflow with the ```trackingOnly``` sequence, previously running in scenario D17, and it looks like the sequence is not yet adapted to the MTD inclusion:

```
----- Begin Fatal Exception 12-Jul-2019 11:15:54 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 2 event: 104 stream: 2
   [1] Running path 'RECOSIMoutput_step'
   [2] Prefetching for module PoolOutputModule/'RECOSIMoutput'
   [3] Prefetching for module RecoChargedRefCandidatePrimaryVertexSorter/'offlinePrimaryVertices4D'
   [4] Prefetching for module ChargedRefCandidateProducer/'trackRefsForJetsBeforeSorting4D'
   [5] Prefetching for module TrackWithVertexRefSelector/'trackWithVertexRefSelectorBeforeSorting4D'
   [6] Prefetching for module PrimaryVertexProducer/'unsortedOfflinePrimaryVertices4D'
   [7] Prefetching for module TOFPIDProducer/'tofPID'
   [8] Calling method for module PrimaryVertexProducer/'unsortedOfflinePrimaryVertices4DnoPID'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: edm::ValueMap&lt;float&gt;
Looking for module label: trackExtenderWithMTD
Looking for productInstanceName: generalTracksigmat0

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```

**Comment by fabiocos** [comment]
assign reconstruction,upgrade

**Comment by kpedro88** [comment]
attn: @makortel @JanFSchulte @mtosi @lgray @bendavid

**Comment by fabiocos** [comment]
The sequence reconstruction_trackingOnly is based on globalreco_tracking https://cmssdt.cern.ch/lxr/source/Configuration/StandardSequences/python/Reconstruction_cff.py#0106 which does not include the module trackExtenderWithMTD, added at the end of globalreco as fastTimingGlobalReco
```
process.globalreco = cms.Sequence(process.globalreco_tracking+process.particleFlowCluster+process.ecalClusters+process.caloTowersRec+process.egammaGlobalReco+process.jetGlobalReco+process.muonGlobalReco+process.pfTrackingGlobalReco+process.muoncosmicreco+process.fastTimingGlobalReco)
```

**Comment by fabiocos** [comment]
the addition of 
```
 
+_phase2_timing_layer_globalreco_tracking = globalreco_tracking.copy()
+_phase2_timing_layer_globalreco_tracking += fastTimingGlobalReco
+(phase2_timing_layer_tile | phase2_timing_layer_bar).toReplaceWith(globalreco_tracking,_phase2_timing_layer_globalreco_tracking)
```
to Reconstruction_cff.py seems to be enough to prevent the failure

**Comment by mtosi** [comment]
thanks !

**Comment by kpedro88** [comment]
Will someone open a PR with the suggestion from @fabiocos? I can also do it, but don't want to duplicate effort

**Comment by perrotta** [comment]
Kevin Pedro <notifications@github.com> ha scritto:

> Will someone open a PR with the suggestion from @fabiocos? I can  
> also do it, but don't want to duplicate effort

Please do.
Thank you Kevin!

**Comment by kpedro88** [comment]
+upgrade

**Comment by perrotta** [comment]
+1
- Apparently solved by #27529
